### PR TITLE
Add user friendly string for calibration timestamp.

### PR DESCRIPTION
**Pull request by dabacon**

### CODE REVIEW

File: cirq/google/engine/calibration.py [hunk 1]
```diff
@@ -96,16 +97,38 @@ def __getitem__(self, key: str) -> Dict[Tuple['cirq.GridQubit', ...], Any]:
...
+
+        Returns:
+            The string in ISO 8601 format YYYY-MM-DDTHH:MM:SS.ffffff.
+        """
+        dt = datetime.datetime.fromtimestamp(self.timestamp / 1000, tz)
+        dt += datetime.timedelta(microseconds=self.timestamp % 1000000)
+        return dt.isoformat(sep=' ', timespec=timespec)
+
     def heatmap(self, key: str) -> vis.Heatmap:
+        """Return a heatmap for metrics that target single qubits.
```
Comment (vtomole) [inline cirq/google/engine/calibration.py]: nit: Too much whitespace in this docstring.

Comment (dabacon) [reply cirq/google/engine/calibration.py]: oh was adding args.  added.

File: cirq/google/engine/calibration.py [hunk 2]
```diff
@@ -96,16 +97,39 @@ def __getitem__(self, key: str) -> Dict[Tuple['cirq.GridQubit', ...], Any]:
...
+            timespec: See datetime.isoformat for valid values.
+
+        Returns:
+            The string in ISO 8601 format YYYY-MM-DDTHH:MM:SS.ffffff.
+        """
+        dt = datetime.datetime.fromtimestamp(self.timestamp / 1000, tz)
+        dt += datetime.timedelta(microseconds=self.timestamp % 1000000)
+        return dt.isoformat(sep=' ', timespec=timespec)
+
     def heatmap(self, key: str) -> vis.Heatmap:
```
Comment (vtomole) [inline cirq/google/engine/calibration.py]: Add a `Returns` docstring too.

Comment (dabacon) [reply cirq/google/engine/calibration.py]: done

File: cirq/google/engine/calibration.py [hunk 3]
```diff
@@ -96,16 +97,46 @@ def __getitem__(self, key: str) -> Dict[Tuple['cirq.GridQubit', ...], Any]:
...
+            tz: The timezone for the string. If None, the method uses the
+                platform's local date and time.
+            timespec: See datetime.isoformat for valid values.
+
+        Returns:
+            The string in ISO 8601 format YYYY-MM-DDTHH:MM:SS.ffffff.
+        """
+        dt = datetime.datetime.fromtimestamp(self.timestamp / 1000, tz)
+        dt += datetime.timedelta(microseconds=self.timestamp % 1000000)
+        return dt.isoformat(sep=' ', timespec=timespec)
```
Comment (mpharrigan) [inline cirq/google/engine/calibration.py]: factor out a method that returns `dt`?

Comment (mpharrigan) [reply cirq/google/engine/calibration.py]: why change sep from default `T`? update the example return value

**Pull request merged**